File: go/aoc_2021/internal/day02/day02_test.go
package day02

import "testing"

func BenchmarkDay02_Part1(b *testing.B) {
	for i := 0; i < b.N; i++ {
		d := New()
		d.Open()
		d.Part1()
	}
}

func BenchmarkDay02_Part2(b *testing.B) {

	for i := 0; i < b.N; i++ {
		d := New()
		d.Open()
		d.Part2()
	}
}

File: javascript/src/aoc_2022/tests/day_02.test.mjs
import assert from "node:assert/strict";
import test from "node:test";
import * as day from "../day_02.mjs";

const input = `A Y
B X
C Z`;

test("Day 2 Part 1", () => {
  const result = day.part1(input);
  assert.strictEqual(result, 15);
});

test("Day 2 Part 2", () => {
  const result = day.part2(input);
  assert.strictEqual(result, 12);
});
